modules/custom-post-types/redirector.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+/**
+ * Jetpack Redirector
+ * Simple method to handle redirects in a scalable manner.
+ * Forked from https://vip.wordpress.com/plugins/wpcom-legacy-redirector/
+ *
+ *  Maintain support for preexisting permalinks after importing or otherwise changing permalink structure.
+ *
+ * Some things were removed from the VIP plugin for simplicity:
+ *   * CLI tooling
+ *   * Functionality for adding redirects in wp-admin
+ *
+ * For now, this has no UI. Data entry is bulk-loaded via WXR / import or custom scripts.
+ *
+ * Redirects are stored as a custom post type (`jetpack-redirect`) and use the following fields:
+ *
+ * - post_name for the sha224 hash of the "from" path or URL.
+ *  - we use this column, since it's indexed and queries are super fast.
+ *  - we also use an sha224 just to simplify the storage.
+ * - post_title to store the non-sha224 version of the "from" path.
+ * - one of either:
+ *  - post_parent if we're redirect to a post; or
+ *  - post_excerpt if we're redirecting to an alternate URL.
+ */
+
+class Jetpack_Redirector {
+	const POST_TYPE = 'jetpack-redirect';
```

Comment:
The core seems to prefer underscored post types https://codex.wordpress.org/Post_Types

Should Jetpack follow or not?

Reply:
Good thought, but I'm following the existing pattern from `jetpack-portfolio`, etc.:

https://github.com/Automattic/jetpack/blob/de978b5e46f04d4586f9026578c95a219c0cb86c/modules/custom-post-types/portfolios.php#L4